install/0000_30_machine-api-operator_13_baremetalhost_configmap.yaml

```diff
@@ -0,0 +1,15 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: ironic-bmo-configmap
+data:
+  http_port: "6180"
+  provisioning_interface: "ens3"
+  provisioning_ip: "172.22.0.3/24"
+  dhcp_range: "172.22.0.10,172.22.0.100"
+  deploy_kernel_url: "http://172.22.0.3:6180/images/ironic-python-agent.kernel"
+  deploy_ramdisk_url: "http://172.22.0.3:6180/images/ironic-python-agent.initramfs"
+  ironic_endpoint: "http://172.22.0.3:6385/v1/"
+  ironic_inspector_endpoint: "http://172.22.0.3:5050/v1/"
+  cache_url: "http://172.22.0.1/images"
```

Comment:
Done.

Reply:
This actually can't be hard coded to anything.  The address on that network is not something we can control.

Maybe this config map shouldn't be in this PR at all, and instead should be created by the installer?  That's where we know all of the variable information, including the RHCOS version.  @hardys

Reply:
Yeah we need some way to pass all of this in via the installer - particularly the IPs on the provisioning network since we can't assume that default subnet will never change (even though it's hard-coded in a few places right now).

For the cacheurl, maybe we could auto-detect the default gateway inside the *downloader container instead of passing any value explicitly here (which would work in any situation where the mirror is running on the virthost). 

I think the point @imain is making is that we could potentially use the libvirt default bridge, which is something pretty unlikely to change, provided whatever server we have running listens on all interfaces, and not only the baremetal/provisioning bridges.

Reply:
Actually no, we can't use the libvirt bridge, because it will only work in the case where masters are virtualized :(

Instead we need this to point to the bootstrap VM provisioning NIC (e.g 172.22.0.2 by default, but will need to be configurable via the install-config), then the bootstrap VM can download from the default gateway or some other mirror, and the master nodes where the BMO run need to download from the bootstrap VM while the cluster bootstrapping is ongoing.

Do we know when the BMO will start relative the the bootstrap-complete event?